src/game/bosses/boss_9.c

```diff
@@ -0,0 +1,135 @@
+#include "global.h"
+#include "gba/io_reg.h"
+#include "sprite.h"
+#include "task.h"
+#include "game/bosses/common.h"
+
+typedef struct {
+    void *func;
+    void *ref;
+    u16 unk4;
+} TA53_Data0;
+
+typedef struct {
+    u8 filler0[0x4];
+    s16 unk4[4];
+    s16 unkC[4];
+    u16 unk14;
+    u8 filler16[0xE];
+    s32 qUnk24;
+    s32 qUnk28;
+} TA53_unk1C; /* size: 0x2C */
+
+typedef struct {
+    void *unk0; // func
+    s16 unk4[4];
+    s16 unkC[4];
+    u16 unk14[4];
+    u16 unk1C[4];
+    u16 unk24[4];
+    u16 unk2C;
+    u16 unk2E;
+    s32 unk30;
+    void *unk34; // ref
+    u8 filler38[0x8];
+} TA53_unk48; /* size: 0x40 */
+
+typedef struct {
+    u8 filler0[0x4];
+    u8 unk4;
+    u8 filler5[0x3 + 0x4];
+    u16 unkC;
+    u8 fillerE[0x6];
+    s32 qUnk14; // worldX?
+    s32 qUnk18; // worldY?
+    u8 filler1C[0x4];
+    Sprite spr20;
+    u8 filler50[0x140];
+} TA53_unkA8; /* size: 0x190 */
+
+typedef struct {
+    void *unk0; // function pointer
+    u16 unk4; // some counter
+    u16 unk6;
+    s32 qUnk8;
+    s32 qUnkC;
+    TA53_unkA8 unkA8[3];
+} TA53_unk98; /* size: 0x190 */
+
+typedef struct {
+    /* 0x000 */ u8 filler0[0x12];
+    /* 0x012 */ u16 unk12;
+    /* 0x014 */ u8 filler14[0x8];
+    /* 0x01C */ TA53_unk1C unk1C;
+    /* 0x048 */ TA53_unk48 unk48;
+    /* 0x098 */ TA53_unk98 unk98;
+    /* 0x548 */ u8 filler548[0x26C];
+    /* 0x7B4 */ Sprite spr7B4;
+    /* 0x7E4 */ u8 filler7E4[0x15C];
+} TA53Boss; /* size: 0x940 */
+
+extern s16 gUnknown_080D8A94[4];
+extern s16 gUnknown_080D8AAC[4];
+extern s16 gUnknown_080D8AC4[4];
+extern s16 gUnknown_080D8ADC[4];
+extern s16 gUnknown_080D8AF4[4];
+extern s16 gUnknown_080D8C7C[4];
+extern s16 gUnknown_080D8C94[4];
+extern TA53_Data0 gUnknown_080D8D6C[8];
+extern TA53_Data0 gUnknown_080D8DCC[8];
+
+extern void CreateTrueArea53Boss(void);
+extern void Task_EggmanKidnapsVanilla(void);
+extern void TaskDestructor_TrueArea53BossGfx(struct Task *);
+
+// Used when Vanilla gets captured
+void SetupEggmanKidnapsVanillaTask(void)
+{
+    TA53Boss *boss;
+    TA53_unk1C *unk1C;
+    TA53_unk48 *unk48;
+    u8 i;
+
+    gActiveBossTask = TaskCreate(Task_EggmanKidnapsVanilla, sizeof(TA53Boss), 0x4000, 0,
+                                 TaskDestructor_TrueArea53BossGfx);
+
+    CreateTrueArea53Boss();
```

Comment:
Does this just init the boss then if it's already been allocated in this func?

Reply:
Well, it's named like all other boss creation functions that are referenced in `game/bosses/common.c`'s array.
And both `CreateTrueArea53Boss` and this function allocate the same number of bytes with a `TaskCreate` call each.

Reply:
Right, so `SetupEggmanKidnapsVanillaTask` actually ends up creating 2 instances of the boss? Cos it allocates 1 here, and then creates another task in `CreateTrueArea53Boss`?